src/core/string_set.cc

```diff
@@ -94,6 +94,8 @@ unsigned StringSet::AddBatch(absl::Span<std::string_view> span, uint32_t ttl_sec
       ++res;
       sds field = MakeSetSds(span[i], ttl_sec);
       AddUnique(field, has_ttl, hash[i]);
+    } else if (update_ttl && has_ttl) {
```

Comment:
I am trying to understand why we need to propaget this update_ttl from set_family.cc calls and not just use the flag here?

Reply:
I added this argument because this same code path is used in `SAdd -> OpAdd -> StringSetWrapper::Add ...`. If we use the flag value here directly, it is applied to two different commands.

Although, in the `SADD` code path the value for ttl is `UINT32_MAX` so `has_ttl` is always false, so it is safe to read the flag here and only update the TTL if both `!flag` and `has_ttl` are true.

I can make the change to use the flag here, if that seems better.

Reply:
please do

Reply:
fixed